pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -442,7 +442,7 @@ func (c *Cloud) AddSSHKeyToAllInstances(user string, keyData []byte) error {
 }
 
 // CurrentNodeName returns the name of the current node
-func (c *Cloud) CurrentNodeName(hostname string) (string, error) {
+func (c *Cloud) CurrentNodeName(hostname string) (types.NodeName, error) {
```

Comment:
pedantic nit: i would be cool with more interface checks for the interfaces `Cloud` implements